google/cloud/spanner/internal/base64.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/base64.h"
+#include <climits>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+namespace {
+
+constexpr char kPadding = '=';
+
+// NOLINTNEXTLINE(modernize-avoid-c-arrays)
+constexpr char kIndexToChar[64 + 1] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    "abcdefghijklmnopqrstuvwxyz"
+    "0123456789"
+    "+/";
+
+// NOLINTNEXTLINE(modernize-avoid-c-arrays)
```

Comment:
Curious: Can you avoid this no-lint by using std::array?

Reply:
This one certainly can.  But the previous one doesn't work if you want to initialize with a string literal.  I thought that was enough of a nicety that I simply went for the NOLINT (and because I'm happy with old arrays).  Still, I've changed to std::array, which I'm happy with if you decide you prefer it.